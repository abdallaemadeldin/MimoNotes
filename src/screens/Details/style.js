import { StyleSheet } from 'react-native';

export const style = () => {
    return StyleSheet.create({
        container: {
            flex: 1,
            backgroundColor: '#f1f1f1',
        },
        descriptionStyle: {
            marginVertical: 15,
            fontSize: 16,
            color: '#000',
            marginHorizontal: 10,
            textAlign: 'left',
        },
        priceStyle: {
            fontWeight: 'bold',
            fontSize: 24,
            marginHorizontal: 10,
            color: '#000'
        }
    });
}