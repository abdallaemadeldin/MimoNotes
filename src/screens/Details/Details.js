import React, { memo } from 'react';
import { View, Text } from 'react-native';
import { Toolbar } from 'src/components/toolbar';
import { style } from './style';
import { useDetails } from 'src/hooks';

const Details = () => {
    const { container, descriptionStyle, priceStyle } = style();
    const { description, price } = useDetails();

    return (
        <View style={container}>
            <Toolbar withBack />

            <Text style={descriptionStyle}>{description}</Text>
            <Text style={priceStyle}>{`$${price}`}</Text>
        </View>
    );
}

export default memo(Details);