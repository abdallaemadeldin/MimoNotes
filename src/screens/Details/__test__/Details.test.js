import React from 'react';
import renderer from 'react-test-renderer';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import Details from '../Details';

jest.mock('react-native-linear-gradient', () => 'LinearGradient');
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => 'Icon');

jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useRoute: () => ({
            params: {},
        }),
        useNavigation: () => ({
            navigate: jest.fn(),
            canGoBack: jest.fn(),
            goBack: jest.fn(),
        }),
    };
});

test('renders correctly', () => {
    const tree = renderer.create(
        <SafeAreaProvider initialMetrics={{ insets: { top: 0, bottom: 0, left: 0, right: 0 } }}>
            <Details />
        </SafeAreaProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});