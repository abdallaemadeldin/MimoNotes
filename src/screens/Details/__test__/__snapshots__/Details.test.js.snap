// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#f1f1f1",
        "flex": 1,
      }
    }
  >
    <LinearGradient
      angle={45}
      colors={
        Array [
          "#21094E",
          "#511281",
          "#7952B3",
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 44,
          "justifyContent": "center",
          "paddingTop": null,
          "width": 750,
        }
      }
      useAngle={true}
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 44,
            "justifyContent": "center",
            "left": 10,
            "opacity": 1,
            "paddingTop": 22,
            "position": "absolute",
            "width": 44,
          }
        }
      >
        <Icon
          color="#fff"
          name="chevron-left"
          size={24}
        />
      </View>
    </LinearGradient>
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 16,
          "marginHorizontal": 10,
          "marginVertical": 15,
          "textAlign": "left",
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontSize": 24,
          "fontWeight": "bold",
          "marginHorizontal": 10,
        }
      }
    >
      $undefined
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
